### ISSUE TITLE
kcov: Incorrect coverage reported when using ${var#patterns} replacements

### ISSUE THREAD

**Issue by Mark-E-Hamilton** [SimonKagstrom/kcov#88]
Hi,
I've been pulled away on other tasks, and won't really be doing a log with bash and kcov for a while. I had a little time last  week, however, and tried testing one of our scripts. The following test script demonstrates an issue I encountered. I dont' recall if I might have submitted this issue before.
````
#/bin/bash
fname='.A'
echo ${fname}
fname="${fname#.}"
echo ${fname}
````
The reports shows 100% coverage with 2 instrumented and executed lines. The output demonstrates that all lines were executed, however.  I tried replacing line 4 (the offending line) with a few other patterns:
````
fname="${fname#.*}"
fname="${fname##.*}"
````
Both of these failed the same way. OTOH, replacing it with this pattern was successful.
````
fname="${fname%*.}"
````
It looks like it's having issues with the sharp in the pattern. Thanks.

**Comment by SimonKagstrom** [comment]
Hi!

kcov probably parses the rest of the line as a comment, causing this particular test to fail, but I'll have to look into that.

**Comment by Mark-E-Hamilton** [comment]
I noticed looking at the output today that the coverage information reappeared after processing a number of lines. I fiddled around with my example to figure out why. What I found is that with this test script:
''''
#/bin/bash
fname='.A'
echo ${fname}
fname="${fname#.}"
echo ${fname}
BACKGROUND='&'
eval "(echo Stuff in subshell
       ) ${BACKGROUND}"
echo "Last stuff"
wait
''''
the coverage turns back on again with the third line from the end. Oddly, if I set up the subshell line like this:
''''
eval "(echo Stuff in subshell ) ${BACKGROUND}"
''''
it does not turn the coverage back on.

**Comment by SimonKagstrom** [comment]
Fixed with 9bd85ab. The problem was caused by treating the end of line 4 above as a comment, which due to other checks leads to a non-terminated {.

I'll look into if there is some yacc parser for shell scripts I can use to avoid these issues.
